x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/esql/esql.ts

```diff
@@ -225,6 +226,16 @@ export const esqlExecutor = async ({
           break;
         }
       }
+      if (
+        completeRule.ruleParams.responseActions?.length &&
+        result.createdSignalsCount &&
+        scheduleNotificationResponseActionsService
+      ) {
```

Comment:
Since we don't depend on the output of `scheduleNotificationResponseActionsService`, can we encapsulate this logic in the `scheduleNotificationResponseActionsService` function implementation since it is repeated in the other rule executors?